llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -42,6 +42,25 @@ struct {
   { tgtok::Endif, "endif" },
   { tgtok::Define, "define" }
 };
+
+// Returns true if `MacroName` is a valid macro name. Valid macro names match
+// the regular expression [a-zA-Z_][0-9a-zA-Z_]* (see prepLexMacroName).
+bool IsValidMacroName(StringRef MacroName) {
+  if (MacroName.size() == 0)
+    return false;
+
+  char First = MacroName[0];
+  if (First != '_' && !isalpha(First))
+    return false;
+
+  // Match the rest of the identifier regex: [0-9a-zA-Z_]*
+  for (char Rest : MacroName.drop_front())
+    if (Rest != '_' && !isalpha(Rest) && !isdigit(Rest))
+      return false;
+
+  return true;
+}
+
 } // end anonymous namespace
```

Comment:
https://llvm.org/docs/CodingStandards.html#anonymous-namespaces

> Because of this, we have a simple guideline: make anonymous namespaces as small as possible, and only use them for class declarations.

Reply:
Done.